spec/ndx-MIES.extensions.yaml

```diff
@@ -1,8 +1,256 @@
+    and third row is the tolerance. Columns are the running index. See also https://alleninstitute.github.io/MIES/labnotebook-docs.html.'
+- neurodata_type_def: LabNotebookTextualValues
+  neurodata_type_inc: NWBData
+  shape:
+  - null
+  - null
+  - null
+  doc: 'Textual labnotebook values: Rows are the running index, Columns hold the different
+    entry names, Layers (up to nine) hold the headstage dependent data in the first
+    8 and the headstage independent data in the 9th layer. See also https://alleninstitute.github.io/MIES/labnotebook-docs.html.'
+- neurodata_type_def: LabNotebookTextualKeys
+  neurodata_type_inc: NWBData
+  shape:
+  - 3
+  - null
+  doc: 'Textual labnotebook keys: First row is the name, second row is the unit and
+    third row is the tolerance. Columns are the running index. See also https://alleninstitute.github.io/MIES/labnotebook-docs.html.'
+- neurodata_type_def: StimulusSetWaveform
+  neurodata_type_inc: NWBData
+  shape:
+  - null
+  - null
+  doc: Stimulus set waveform data. This is only present if not all three parameter
+    waves could be found or a third-party stimset was used. One column per sweep.
+  quantity: '?'
+- neurodata_type_def: StimulusSetWavebuilderParameter
+  neurodata_type_inc: NWBData
+  shape:
+  - null
+  - 100
+  - 9
+  doc: Numerical part of the stimulus set parameter waves for recreating the stimset
+    in MIES. Rows are the data entries, Columns are the index of the segment/epoch
+    and Layers hold different stimulus waveform types. See also https://alleninstitute.github.io/MIES/file/_m_i_e_s___wave_data_folder_getters_8ipf.html#_CPPv423GetWaveBuilderWaveParamv.
+- neurodata_type_def: StimulusSetWavebuilderParameterText
+  neurodata_type_inc: NWBData
+  shape:
+  - null
+  - 100
+  - 9
+  doc: Textual part of the stimulus set parameter waves for recreating the stimset
+    in MIES. Rows are the data entries, Columns are the index of the segment/epoch
+    (last index holds settings for the full set) and Layers hold different stimulus
+    waveform types. See also https://alleninstitute.github.io/MIES/file/_m_i_e_s___wave_data_folder_getters_8ipf.html#_CPPv427GetWaveBuilderWaveTextParamv.
+- neurodata_type_def: StimulusSetWavebuilderSegmentTypes
+  neurodata_type_inc: NWBData
+  shape:
+  - 102
+  doc: Stimulus set parameters for the full set. See also https://alleninstitute.github.io/MIES/file/_m_i_e_s___wave_data_folder_getters_8ipf.html#_CPPv418GetSegmentTypeWavev.
 groups:
-- neurodata_type_def: TetrodeSeries
-  neurodata_type_inc: ElectricalSeries
-  doc: An extension of ElectricalSeries to include the tetrode ID for each time series.
-  attributes:
-  - name: trode_id
-    dtype: int32
-    doc: The tetrode ID.
+- neurodata_type_def: UserComment
+  doc: Free form text notes from the experimenter
+  groups:
+  - neurodata_type_inc: UserCommentDevice
+    doc: Device for the user text notes
+    datasets:
```

Comment:
The definition of `UserCommentDevice` already specifies that it includes a `UserCommentString` dataset, so this is unnecessary.

Reply:
Thanks, you fixed that already in the create_spec_extension.py file. But I think I now understand how that works. I knew yesterday that this is way to verbose.